samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -240,24 +329,26 @@ static struct bt_mesh_sensor *const sensors[] = {
 static struct bt_mesh_sensor_srv sensor_srv =
 	BT_MESH_SENSOR_SRV_INIT(sensors, ARRAY_SIZE(sensors));
 
-static struct k_work_delayable end_of_presence_work;
+static struct k_work_delayable presence_detected_work;
 
-static void end_of_presence(struct k_work *work)
+static void presence_detected(struct k_work *work)
 {
 	int err;
 
 	/* This sensor value must be boolean -
 	 * .val1 can only be '0' or '1'
 	 */
 	struct sensor_value val = {
-		.val1 = 0,
+		.val1 = 1,
 	};
 
 	err = bt_mesh_sensor_srv_pub(&sensor_srv, NULL, &presence_sensor, &val);
 
 	if (err) {
 		printk("Error publishing end of presence (%d)\n", err);
 	}
+
+	prev_pres = 0;
```

Comment:
When detection is done it will report 0 as it should do. So it need to clear the prev_detection.

Reply:
Umm, why do you think so? When detection is done, it should keep reporting the time since last known presence. That is what the property is.  

> 3.9.6 Time Since Presence Detected property 
This property represents the time that has elapsed since the sensor last detected presence. This value can, for example, be used to turn off lights in the absence of any activity in a space.

Reply:
This will work like this:

1. Send Sensor Get to fetch "Time since presence detected". Received response will be x seconds since last detection.
2. Press and hold button to detected the presence. Detection presence.
3. Wait for 60 seconds.
4. Send Sensor Get to fetch "Time since presence detected". Received response will be 0 seconds since last detection as presence is active.
5. Wait for 60 seconds.
6. Release button to represent no presence presence. No presence.
7. Send Sensor Get to fetch "Time since presence detected". Received response will be x seconds since last detection.

The question is last present running from when presence is started detected, or when presence is no longer active. I would assume when the "Time since present detected" is from when presence is no longer present. If not then the time for something moving will be added to the time, and that is a little bit strange to add.
I may give the effect that if someplace has a movement over time and "Time since presence detected" is ex. 5 min it will turn of the light, instead of after 5 min without movement.

Reply:
Ok, this looks fine then.